tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -345,7 +352,7 @@ public DAGAppMaster(ApplicationAttemptId applicationAttemptId,
       ContainerId containerId, String nmHost, int nmPort, int nmHttpPort,
       Clock clock, long appSubmitTime, boolean isSession, String workingDirectory,
       String [] localDirs, String[] logDirs, String clientVersion,
-      Credentials credentials, String jobUserName, AMPluginDescriptorProto pluginDescriptorProto) {
+      Credentials credentials, String jobUserName, AMPluginDescriptorProto pluginDescriptorProto, Configuration conf) {
```

Comment:
I don't think we need a configuration here: as far as I can see the patch is about reading the default configuration and creating thread dumps from AM start to end, which is suboptimal (we don't really care about thread dumps of an idle TezAM), I was thinking about this feature a bit differently:

1. we should only make thread dumps while running a DAG, so need to start dumps in startDAG and stop dumps at the end
2. we should rely on dag config (newDAG.getConf()), this way we'll make this feature easily configurable on the fly, as the DAG's config is supposed to contain on-the-fly session props, e.g. when a hive user simply does this in beeline:
```
set tez.job.periodic.thread.dump.frequency=5;
```

Reply:
Thanx for the pointers Laszlo, I tried a new approach as you suggested.